control-plane/csi-controller/src/server.rs

```diff
@@ -0,0 +1,126 @@
+use crate::MayastorApiClient;
+use futures::TryFutureExt;
+use tokio::{
+    io::{AsyncRead, AsyncWrite, ReadBuf},
+    net::UnixListener,
+};
+use tonic::transport::{server::Connected, Server};
+
+use std::{
+    fs,
+    io::ErrorKind,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use rpc::csi::{controller_server::ControllerServer, identity_server::IdentityServer};
+
+use crate::{controller::CsiControllerSvc, identity::CsiIdentitySvc};
+
+#[derive(Debug)]
+struct UnixStream(pub tokio::net::UnixStream);
+
+impl Connected for UnixStream {
+    type ConnectInfo = UdsConnectInfo;
+
+    fn connect_info(&self) -> Self::ConnectInfo {
+        UdsConnectInfo {
+            peer_addr: self.0.peer_addr().ok().map(Arc::new),
+            peer_cred: self.0.peer_cred().ok(),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct UdsConnectInfo {
+    pub peer_addr: Option<Arc<tokio::net::unix::SocketAddr>>,
+    pub peer_cred: Option<tokio::net::unix::UCred>,
+}
+
+impl AsyncRead for UnixStream {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<std::io::Result<()>> {
+        Pin::new(&mut self.0).poll_read(cx, buf)
+    }
+}
+
+impl AsyncWrite for UnixStream {
+    fn poll_write(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &[u8],
+    ) -> Poll<std::io::Result<usize>> {
+        Pin::new(&mut self.0).poll_write(cx, buf)
+    }
+
+    fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<std::io::Result<()>> {
+        Pin::new(&mut self.0).poll_flush(cx)
+    }
+
+    fn poll_shutdown(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<std::io::Result<()>> {
+        Pin::new(&mut self.0).poll_shutdown(cx)
+    }
+}
+
+pub struct CsiServer {}
+
+async fn ping_rest_api() {
+    info!("Checking REST API endpoint accessibility ...");
+
+    match MayastorApiClient::get_client().list_nodes().await {
```

Comment:
We should probably add health endpoints to the rest api

Reply:
I captured that in CAS-1125